Library/Homebrew/software_spec.rb

```diff
@@ -208,6 +208,7 @@ def recursive_requirements
   end
 
   def patch(strip = :p1, src = nil, &block)
+    dependency_collector.add("gpatch" => :build) unless which("patch")
```

Comment:
I'd like to see this go in a more appropriate location e.g. in `dependency_collector.rb` which could be called by this method if necessary with something like `dependency_collector#gpatch_dep_if_needed` which is called by `dependency_collector.add_gpatch_dep_if_needed` or similar.

Reply:
@MikeMcQuaid I'm afraid I've already exhausted my understanding of Ruby at this point.  I'm happy to fiddle around with this further but it will: A) Be a while and B) Pure hackery ;)

Reply:
That's fine. Another @Homebrew/linux person may wish to commit directly to your PR. Feel free to ask any questions of anyone here, too, and we'll do our best to help. Thanks again!

Reply:
Plenty of Homebrew developers and even maintainers got their start on Ruby programming with Homebrew, like myself for example! Since this isn't a pressing bug, and it's also a good first issue, I'd be happy to help you work your way through it, even if it takes some time. I'll comment with suggestions below.

Reply:
แนะนำผมด้วยครับ มีอะไรที่ต้องแก้ไขเพิ่มเติม